src/api/SearchApi.js

```diff
@@ -74,39 +65,53 @@ class RequestSerializer {
   };
 }
 
-/** Default backend response serializer */
-class ResponseSerializer {
-  _serializeAggregation(bucket) {
-    const aggregation = {
-      key: bucket.key,
-      total: bucket.doc_count,
-      hasNestedField: false,
-    };
+const extractFromPath = (aggregation, pathToExtract) => {
+  return pathToExtract.reduce((obj, key) =>
+    (obj && obj[key] !== 'undefined') ? obj[key] : undefined, aggregation);
+}
+
+const createAggregation = (parentKeyName, buckets) => {
+  let aggregations = new Object();
 
+  buckets.forEach(function (item) {
     const nestedField = _find(
-      Object.keys(bucket),
+      Object.keys(item),
       key =>
-        _isPlainObject(bucket[key]) &&
-        'buckets' in bucket[key] &&
-        bucket[key].buckets.length
+        _isPlainObject(item[key]) &&
+        'buckets' in item[key] &&
+        item[key].buckets.length
     );
     if (nestedField) {
-      const buckets = bucket[nestedField].buckets;
-      aggregation['hasNestedField'] = nestedField;
-      aggregation[nestedField] = buckets.map(bucket =>
-        this._serializeAggregation(bucket)
-      );
+      const nestedAggregationPath = [nestedField, "buckets"];
+      const nestedBuckets = extractFromPath(item, nestedAggregationPath);
+      const newKeyName = "".concat(parentKeyName, ".", item.key, '.', nestedField);
+      aggregations[item.key] = createAggregation(newKeyName, nestedBuckets, nestedAggregationPath);
+      aggregations[item.key]["key"] = newKeyName;
+      aggregations[item.key]["name"] = item.key;
+      aggregations[item.key]["total"] = item.doc_count;
+      aggregations[item.key]["hasNestedField"] = nestedField;
     }
-    return aggregation;
-  }
+    else {
+      delete item.subtype;
+      aggregations[item.key] = item;
```

Comment:
I think you can delete this line no? Why not just:

```
const item = {
    'hasNestedField': false,
    'name': item.key,
    'total': item.doc_count,
    'key': parentKeyName.concat(".", item.key)
}
aggregations[item.key] = item;
```
But I might miss something here... the values maybe?